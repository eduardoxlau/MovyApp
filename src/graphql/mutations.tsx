import { gql } from '@apollo/client';

export const LOGIN = gql`
  mutation Login($input: LoginInput!) {
    login(input: $input) {
      access_token
      user {
        id
        email
        full_name
        admin
        photo_path
      }
    }
  }
`;
export const CREATE_USER = gql`
  mutation CreateUser($input: CreateUserInput!) {
    createUser(input: $input) {
      id
      email
      full_name
      admin
      photo_path
    }
  }
`;

export const UPDATE_USER = gql`
  mutation UpdateUser($input: UpdateUserInput!) {
    updateUser(input: $input) {
      status
    }
  }
`;

export const REMOVE_USER = gql`
  mutation DeleteUser {
    deleteUser {
      status
    }
  }
`;

export const CREATE_LIST = gql`
  mutation CreateList($input: CreateListInput!) {
    createList(input: $input) {
      id
      name
      description
    }
  }
`;

export const EDIT_LIST = gql`
  mutation UpdateList($input: UpdateListInput!) {
    updateList(input: $input) {
      status
    }
  }
`;

export const REMOVE_LIST = gql`
  mutation DeleteList($id: Float!) {
    deleteList(id: $id) {
      status
    }
  }
`;

export const REMOVE_MOVIE_LIST = gql`
  mutation RemoveMovieToList($input: ListMovieInput!) {
    removeMovieToList(input: $input) {
      status
    }
  }
`;

export const ADD_MOVIE_LIST = gql`
  mutation AddMovieToList($input: ListMovieInput!) {
    addMovieToList(input: $input) {
      status
    }
  }
`;

export const SEEN_MOVIE = gql`
  mutation seenMovie($movieId: Float!) {
    seenMovie(movieId: $movieId) {
      status
    }
  }
`;
