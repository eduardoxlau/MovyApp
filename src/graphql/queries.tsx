import { gql } from '@apollo/client';

export const GET_USER = gql`
  query GetUser {
    getUser {
      id
      email
      full_name
      admin
      lists {
        description
        movies {
          id
          title
        }
      }
    }
  }
`;

export const GET_MOVIES_SEEN = gql`
  query GetMoviesSeen {
    getMoviesSeen {
      id
      title
      overview
      trailer_url
      card_url
      poster_url
      backdrop_url
      stars
      genres {
        name
      }
    }
  }
`;

export const GET_MOVIE = gql`
  query GetMovie($id: Float!) {
    getMovie(id: $id) {
      id
      title
      overview
      trailer_url
      card_url
      poster_url
      backdrop_url
      stars
      genres {
        name
      }
    }
  }
`;

export const GET_MOVIES = gql`
  query GetMovies {
    getMovies(limit: 20) {
      items {
        id
        title
        overview
        trailer_url
        card_url
        poster_url
        backdrop_url
        stars
        genres {
          name
        }
      }
    }
  }
`;

export const GET_MOVIES_PAGINATE = gql`
  query GetMovies($page: Float, $limit: Float) {
    getMovies(page: $page, limit: $limit) {
      meta {
        totalPages
        currentPage
      }
      items {
        id
        title
        overview
        trailer_url
        card_url
        poster_url
        backdrop_url
        stars
        genres {
          name
        }
      }
    }
  }
`;

export const GET_LISTS = gql`
  query GetLists {
    getLists {
      id
      name
      description
      movies {
        id
        title
        trailer_url
        card_url
      }
    }
  }
`;
