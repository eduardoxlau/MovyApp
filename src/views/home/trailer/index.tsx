import { Link } from 'react-router-dom';

import Star from 'assets/icons/star.png';
import Play from 'assets/icons/play.png';
import Imdb from 'assets/icons/imdb.png';
import Save from 'assets/icons/save.png';
import Sound from 'assets/icons/sound.png';
import Camera from 'assets/icons/camera.png';
import { ItemInterface } from 'components/card';

const Trailer = ({
  backdrop_url,
  title,
  overview,
  stars,
  id,
}: ItemInterface) => (
  <div
    className="w-full relative bg-cover bg-top bg-center flex flex-col"
    style={{
      backgroundImage: `url("${backdrop_url}")`,
      minHeight: '600px',
    }}
  >
    <div className="w-full h-full absolute bg-black opacity-50 z-0" />
    <div className="w-full h-52 bglinear-black-180" />
    <div className="container flex flex-col md:w-4/5 mx-auto flex-grow z-10">
      <div className="py-6 opacity-60">2019 . Action, Thiller</div>
      <div className="flex items-center flex-col md:flex-row">
        <div className="text-6xl font-black opacity-80 mb-2">{title}</div>
        <div className="rounded-full border border-white mr-auto md:mr-0 md:ml-auto text-xl px-6 py-1 font-bold opacity-90">
          PG13
        </div>
      </div>

      <div className="flex my-8 items-center justify-around md:justify-start">
        <div
          className="bg-pink-600 px-2 py-0.5 rounded md:mr-20 text-sm flex items-center"
          style={{ height: 'fit-content' }}
        >
          2.30h
        </div>
        <div className="mx-10 md:mr-20 text-lg flex items-center text-center flex-col md:flex-row">
          <img className="w-6 mr-3" src={Camera} alt="" />
          <div>Quentien cuarentino</div>
        </div>
        <div className="flex items-center">
          {Array.from(Array(stars).keys()).map((_) => (
            <img className="w-4 mr-1" src={Star} alt="" key={_} />
          ))}
        </div>
      </div>
      <div className="mb-40 mt-10 text-lg leading-8">{overview}</div>
      <div className="flex flex-grow items-center text-lg color-trailer flex-col md:flex-row">
        <div className="flex font-thin">
          <div className="flex items-center">
            <img className="w-5 mr-3" src={Save} alt="" />
            Watch Later
          </div>
          <div className="flex items-center">
            <img className="w-5 md:ml-20 mr-3" src={Play} alt="" />
            <Link to={`/trailer/${id}`}>Watch Trailer</Link>
          </div>
          <div className="flex items-center">
            <img className="w-20 md:ml-20 mr-3" src={Imdb} alt="" />
          </div>
        </div>

        <div className="mt-8 md:mt-0 mx-auto md:mr-0 md:ml-auto">
          <img className="w-8" src={Sound} alt="" />
        </div>
      </div>
    </div>
    <div className="w-full h-32 bglinear-black" />
  </div>
);

export default Trailer;
