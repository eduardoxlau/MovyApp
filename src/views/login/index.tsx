import { LOGIN } from 'graphql/mutations';
import { useMutation } from '@apollo/client';
import { useHistory } from 'react-router-dom';
import { ChangeEvent, FormEvent, useState, useEffect, useContext } from 'react';

import Fb from 'assets/icons/fb.png';
import Input from 'components/input/light';
import { UserContext } from 'storage/context';
import Notification from 'components/notification';

const Login = () => {
  const { setContext } = useContext(UserContext);
  const history = useHistory();

  const [login, { data, loading, error }] = useMutation(LOGIN, {
    errorPolicy: 'all',
  });

  useEffect(() => {
    if (data) {
      const {
        login: { access_token, user },
      } = data;
      setContext({ access_token, isAuth: true, user });
      history.push('/home');
    }
  }, [data]);

  const [formData, updateFormData] = useState({});

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    updateFormData({
      ...formData,
      [e.target.name]: e.target.value,
    });
  };
  const onSubmit = async (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    login({ variables: { input: formData } });
  };
  return (
    <div className="flex flex-col bg-logo bg-cover">
      <form onSubmit={onSubmit}>
        <div className="container flex-grow flex-shrink mx-auto mt-44">
          <div className="bg-black text-white m-auto px-12 py-12 w-auto md:w-login bg-opacity-80 rounded">
            <div className="text-4xl text-center">Inicia sesión</div>
            <div className="my-9">
              <Input
                placeholder="Email o número de teléfono"
                name="email"
                type="email"
                required
                onChange={handleChange}
              />
              <Input
                placeholder="Contraseña"
                name="password"
                type="password"
                required
                onChange={handleChange}
              />
            </div>
            {error && <Notification message={error.message} />}
            <input
              disabled={loading}
              type="submit"
              className={`w-full h-11 text-center rounded cursor-pointer ${
                loading ? 'bg-blue-300' : 'bg-button'
              }`}
              value="Iniciar sesión"
            />
            <div className="flex items-center my-3">
              <div className="w-1/2 flex">
                <div className="flex items-center">
                  <input
                    type="checkbox"
                    className="mr-5 w-5 h-5 bg-coolGray-400"
                  />
                </div>
                Recuerdame
              </div>
              <div className="w-1/2 flex justify-end">Necesitas ayuda?</div>
            </div>
            <div className="my-10 flex justify-center items-center">
              <img src={Fb} alt="" className="mr-4" />
              Iniciar sesión con Facebook
            </div>
            <div className="text-center">
              ¿Primera vez en Movy?
              <span className="font-bold">Suscríbete ya.</span>
            </div>
          </div>
        </div>
        <div className="h-40 bglinear-black" />
      </form>
    </div>
  );
};

export default Login;
