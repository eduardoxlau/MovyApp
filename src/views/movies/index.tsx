import { useState } from 'react';
import { useQuery } from '@apollo/client';

import { GET_MOVIES_PAGINATE } from 'graphql/queries';
import { ItemInterface } from 'components/card';

const Movies = () => {
  const [page, setPage] = useState(1);
  let movies: ItemInterface[] = [];
  let pages = 1;
  const { data } = useQuery(GET_MOVIES_PAGINATE, {
    variables: {
      page,
      limit: 8,
    },
  });

  if (data) {
    pages = data.getMovies.meta.totalPages;
    movies = data.getMovies.items;
  }

  return (
    <div className="bg-black flex flex-col flex-grow">
      <div className="container mx-auto mt-40 mb-20">
        <div className="flex w-full flex-col md:flex-row text-white flex-wrap">
          {movies?.map((movie: ItemInterface) => (
            <div className="flex flex-col px-3 w-full md:w-1/4 mb-10">
              <img className="h-50 w-full" src={movie.card_url} alt="" />
              <div className="flex my-4 items-center">
                <div className="text-xl">{movie.title}</div>
                <div className="rounded-full border border-white ml-auto text-lg px-4 py-0 font-bold opacity-90">
                  PG13
                </div>
              </div>
              <div>{`${movie.overview.slice(0, 100)} ...`}</div>
            </div>
          ))}
        </div>
        <div className="flex justify-center text-center text-white">
          {Array(pages)
            .fill('')
            .map((_, index) => (
              <div
                className={`m-1 px-2 py-1 rounded cursor-pointer ${
                  page === index + 1 && 'bg-blue-500'
                }`}
                onClick={() => setPage(index + 1)}
              >
                {index + 1}
              </div>
            ))}
        </div>
      </div>
    </div>
  );
};

export default Movies;
