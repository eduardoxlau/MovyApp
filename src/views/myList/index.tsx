import { Link } from 'react-router-dom';
import { useQuery } from '@apollo/client';

import Loading from 'components/loading';
import { GET_LISTS } from 'graphql/queries';
import Card, { ItemInterface } from 'components/card';

const MyList = () => {
  const { data, loading } = useQuery(GET_LISTS);

  if (loading) return <Loading />;

  const lists: { name: string; movies: ItemInterface[] }[] = data.getLists;

  return (
    <div className="bg-black flex flex-col flex-grow">
      <div className="container mx-auto mt-20">
        {lists.map(({ name, movies }) => (
          <div className="text-white mt-11">
            <div className="text-3xl">{name}</div>
            <div className="flex my-7 w-full overflow-scroll items-center container-card">
              {movies.map((item: ItemInterface, index: number) => (
                <div className="mx-1 cursor-pointer">
                  <Link to={`/trailer/${item.id}`}>
                    <Card item={item} scale index={index} />
                  </Link>
                </div>
              ))}
            </div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default MyList;
