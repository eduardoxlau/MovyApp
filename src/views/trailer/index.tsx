import { useState } from 'react';
import { useMutation, useQuery } from '@apollo/client';

import Loading from 'components/loading';

import {
  ADD_MOVIE_LIST,
  REMOVE_MOVIE_LIST,
  SEEN_MOVIE,
} from 'graphql/mutations';
import Modal from 'components/modal';
import Add from 'assets/icons/add.png';
import Star from 'assets/icons/star.png';
import Save from 'assets/icons/save.png';
import Play from 'assets/icons/bt_play.png';
import Close from 'assets/icons/close.png';
import Camera from 'assets/icons/camera.png';
import { useParams } from 'react-router-dom';
import { ItemInterface } from 'components/card';
import { GET_MOVIE, GET_LISTS } from 'graphql/queries';
import { ReadFieldFunction } from '@apollo/client/cache/core/types/common';

export type List = {
  id: number;
  name: string;
  description: string;
  public: boolean;
  movies: ItemInterface[];
};

const Trailer = () => {
  const [infoModal, setInfoModal] = useState<{
    show: boolean;
    list: List | null;
    isEdit: boolean;
  }>({
    show: false,
    list: null,
    isEdit: false,
  });

  const { id } = useParams<{ id: string }>();
  const [isOpenList, openList] = useState(false);
  const [isPlay, setPlay] = useState(false);

  const { data, loading } = useQuery(GET_MOVIE, {
    variables: { id: parseInt(id, 10) },
  });

  const { data: dataLists, loading: loadingLists } = useQuery(GET_LISTS);

  const lists: List[] = dataLists?.getLists;

  const [removeMovie, { loading: loadingRemoveMovie }] = useMutation(
    REMOVE_MOVIE_LIST,
    {
      errorPolicy: 'all',
    }
  );

  const [playMovie] = useMutation(SEEN_MOVIE, {
    errorPolicy: 'all',
  });

  const [addMovie, { loading: loadingAddMovie }] = useMutation(ADD_MOVIE_LIST, {
    errorPolicy: 'all',
  });

  if (loading || loadingLists) return <Loading />;

  const movie: ItemInterface = data.getMovie;

  const list = lists.find(({ movies }) =>
    movies.some(({ id: movieId }) => movieId === movie.id)
  );

  const updateListCache = ({
    listsCache,
    readField,
    listMutated,
  }: {
    listsCache: List[];
    readField: ReadFieldFunction;
    listMutated: List;
  }) =>
    listsCache.reduce((acc: List[], listCurr: List) => {
      let newList = { ...listCurr };
      if (readField('id', newList) === listMutated.id) {
        newList = listMutated;
      }
      acc.push(newList);
      return acc;
    }, []);

  const addMovieToList = async (listId: number) => {
    if (!loadingAddMovie) {
      addMovie({
        variables: { input: { listId, movieId: movie.id } },
        update(cache, { data: { addMovieToList: listMutated } }) {
          cache.modify({
            fields: {
              getLists(listsCache: List[], { readField }) {
                return updateListCache({ listsCache, readField, listMutated });
              },
            },
          });
        },
      });
    }
  };

  const removeMovieToList = () => {
    if (list && !loadingRemoveMovie) {
      removeMovie({
        variables: { input: { listId: list.id, movieId: movie.id } },
        update(cache, { data: { removeMovieToList: listMutated } }) {
          cache.modify({
            fields: {
              getLists(listsCache: List[], { readField }) {
                return updateListCache({ listsCache, readField, listMutated });
              },
            },
          });
        },
      });
    }
  };

  const playVideo = async () => {
    await playMovie({
      variables: { movieId: movie.id },
    });
    setPlay(true);
  };

  return (
    <div className="bg-black flex flex-col flex-grow text-white">
      <div className="container mx-auto my-40">
        <div className="flex mb-4 flex-col md:flex-row">
          <div className="flex">
            <div>2019 . Action, Thiller</div>
            <div className="flex items-center ml-20">
              <img className="w-5 mr-3" src={Save} alt="" />
              Watch Later
            </div>
          </div>
          <div className="flex items-center mt-3 md:mt-0 md:ml-20 relative  ">
            {list ? (
              <div
                className="cursor-pointer flex"
                onClick={() => removeMovieToList()}
              >
                <img className="w-7  mr-3" src={Close} alt="" />
                Remove to list
              </div>
            ) : (
              <div onClick={() => openList((state) => !state)}>
                <div className="flex cursor-pointer">
                  <img className="w-7 h-7 mr-3" src={Add} alt="" />
                  add to list
                </div>
                <div
                  onMouseLeave={() => openList(false)}
                  className={`flex flex-col bg-blue-500 p-3 rounded top-8 min-w-full w-max text-center ${
                    isOpenList ? 'absolute' : 'hidden'
                  }`}
                >
                  {lists.map(({ name, id: listId }) => (
                    <div
                      className="cursor-pointer hover:font-bold"
                      onClick={() => addMovieToList(listId)}
                      key={listId}
                    >
                      {name}
                    </div>
                  ))}
                  <div
                    className="cursor-pointer font-bold  text-lg"
                    onClick={() => {
                      setInfoModal((prev) => ({
                        ...prev,
                        show: true,
                      }));
                    }}
                  >
                    Create new List
                  </div>
                </div>
              </div>
            )}
          </div>
        </div>

        <div className="text-6xl font-black ">{movie?.title}</div>
        <div className="flex mt-3 mb-16 items-center justify-around md:justify-start">
          <div
            className="bg-pink-600 px-2 py-0.5 rounded md:mr-5 text-sm flex items-center"
            style={{ height: 'fit-content' }}
          >
            2.30h
          </div>
          <div className="mx-10 md:mr-6 text-lg flex items-center text-center flex-col md:flex-row">
            <img className="w-6 mr-3" src={Camera} alt="" />
            <div>Quentien cuarentino</div>
          </div>
          <div className="flex items-center">
            {Array.from(Array(movie.stars).keys()).map((_) => (
              <img className="w-4 mr-1" src={Star} alt="" key={_} />
            ))}
          </div>
        </div>
        <div
          className="w-2/3 overflow-hidden relative m-auto "
          style={{ height: '500px' }}
        >
          {movie?.trailer_url && (
            <iframe
              width="100%"
              className="w-full h-full absolute"
              height="auto"
              src={
                isPlay ? `${movie?.trailer_url}?autoplay=1` : movie?.trailer_url
              }
            />
          )}
          {!isPlay && (
            <>
              <div className="absolute w-full h-full bg-black opacity-50" />
              <div className="absolute w-full h-full  flex justify-center items-center opacity-80">
                <img
                  className="w-40 cursor-pointer"
                  src={Play}
                  alt=""
                  onClick={playVideo}
                />
              </div>
            </>
          )}
        </div>
      </div>
      {infoModal.show && (
        <Modal
          show={infoModal.show}
          edit={infoModal.isEdit}
          list={infoModal.list}
          onClose={() => {
            setInfoModal((prev) => ({
              ...prev,
              show: false,
            }));
          }}
        />
      )}
    </div>
  );
};

export default Trailer;
