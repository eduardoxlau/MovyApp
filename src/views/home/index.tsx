/* eslint-disable comma-dangle */
import { useState } from 'react';
import Card, { ItemInterface } from 'components/card';

import { movies, moviesByMovy } from 'mocks/movies';

import Trailer from './trailer';
import Preview from './preview';
import About from './about';

type CardProps = {
  title: string;
  items: ItemInterface[];
  showPercentage?: boolean;
  type?: 'wide' | 'large';
  scale?: boolean;
  border?: boolean;
};
const Login = () => {
  const [currentItem, setItem] = useState<ItemInterface>(movies[0]);
  const sectionCards = ({
    title,
    items,
    showPercentage = false,
    type = 'wide',
    scale = true,
    border = false,
  }: CardProps) => {
    const onClickCard = (item: ItemInterface) => setItem(item);

    return (
      <div>
        <div className="container text-2xl mx-auto">{title}</div>
        <div
          className={`flex mt-1 pb-6 relative  w-full container-movies overflow-x-scroll overflow-y-hidden  items-center container-card ${
            type === 'large' ? 'p-6' : 'py-0'
          }`}
        >
          {items.map((item, index) => (
            <div className="mx-1 z-0" key={item.id}>
              <Card
                item={item}
                scale={scale}
                type={type}
                index={index}
                onSelected={onClickCard}
                border={border}
                idSelected={currentItem?.id}
              />
              {showPercentage && (
                <div className="relative w-full h-1.5 bg-gray-200 rounded">
                  <div
                    className="absolute h-full bg-blue-500 rounded"
                    style={{ width: `${Math.random() * 100}%` }}
                  />
                </div>
              )}
            </div>
          ))}
        </div>
      </div>
    );
  };

  return (
    <div className="bg-black flex flex-col text-white">
      <Trailer />
      {sectionCards({ title: 'My List', items: movies })}
      {sectionCards({ title: 'Popular on Movy', items: movies })}
      {sectionCards({
        title: 'Continue Watching for John',
        items: movies,
        showPercentage: true,
      })}
      <Preview />
      {sectionCards({
        title: 'Most Viewed',
        items: moviesByMovy,
        type: 'large',
      })}
      {sectionCards({
        title: 'Recommended movies',
        items: movies,
        border: true,
        scale: false,
      })}
      <About {...currentItem} />
    </div>
  );
};

export default Login;
