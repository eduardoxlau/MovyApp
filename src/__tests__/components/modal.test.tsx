/* eslint-disable no-undef */
import wait from 'waait';
import { render, fireEvent } from '@testing-library/react';

import { MockedProvider } from '@apollo/client/testing';

import { GET_LISTS } from 'graphql/queries';
import { EDIT_LIST } from 'graphql/mutations';
import Modal from 'components/modal';
import { List } from 'views/trailer';
import ListMock from '__mocks__/lists.mock';

const list: List = ListMock[0];

const mocks = [
  {
    request: {
      query: EDIT_LIST,
      variables: {
        input: {
          id: 8,
          name: "Rafael's list",
          description: 'Description list',
        },
      },
    },
    result: {
      data: {
        updateList: {
          status: 'success',
        },
      },
    },
  },
  {
    request: {
      query: GET_LISTS,
    },
    result: { data: { getLists: [list] } },
    newData: () => ({ data: { getLists: [list] } }),
  },
];

describe('<Modal />', () => {
  it('should render snapshot Modal', async () => {
    const { asFragment } = render(
      <MockedProvider mocks={mocks}>
        <Modal />
      </MockedProvider>
    );
    await wait(0);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should submit form', async () => {
    const onClose = jest.fn();
    const { getByText } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Modal onClose={onClose} edit list={list} />
      </MockedProvider>
    );
    fireEvent.click(getByText(/Save/i));
    await wait(200);
    expect(onClose).toHaveBeenCalled();
  });
});
