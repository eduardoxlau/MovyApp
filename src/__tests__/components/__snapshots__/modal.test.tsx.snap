// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render snapshot Modal 1`] = `
<DocumentFragment>
  <div
    class="w-full h-screen flex items-center fixed"
  >
    <div
      class="z-30 bg-black w-full md:w-1/2  min-h-80 m-auto text-white p-10 relative"
    >
      <img
        alt=""
        class="w-8 absolute top-2 right-2 cursor-pointer"
        src="close_window.png"
      />
      <form
        class="flex flex-col"
      >
        <div
          class="text-3xl text-center"
        >
          Create  List
        </div>
        <div>
          <div>
            Name:
          </div>
          <input
            class="rounded w-full h-10 my-1.5 text-black indent-1"
            name="name"
            placeholder=""
            required=""
            type="text"
            value=""
          />
        </div>
        <div>
          <div>
            Description:
          </div>
          <input
            class="rounded w-full h-10 my-1.5 text-black indent-1"
            name="description"
            placeholder=""
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="flex items-center mt-2"
        >
          <div>
            Is public?
          </div>
          <input
            class="ml-5 w-8 h-8 order-1 md:order-none"
            name="isKid"
            type="checkbox"
          />
        </div>
        <div
          class="mt-5"
        >
          <input
            class="w-full h-11 text-center rounded cursor-pointer bg-button"
            type="submit"
            value="Save"
          />
        </div>
      </form>
    </div>
    <div
      class="z-20 absolute opacity-60 bg-black w-full h-full"
    />
  </div>
</DocumentFragment>
`;
